spec/system/users/contact_creation_spec.rb

```diff
@@ -1,20 +1,38 @@
-# spec/system/contact_creation_spec.rb
-require 'rails_helper'
-
 RSpec.describe "Contact creation", type: :system do
+  let(:english_user) { FactoryBot.create(:user, user_setting: FactoryBot.create(:user_setting, preferred_language: 'en')) }
+  let(:german_user) { FactoryBot.create(:user, user_setting: FactoryBot.create(:user_setting, preferred_language: 'de')) }
+  let(:spanish_user) { FactoryBot.create(:user, user_setting: FactoryBot.create(:user_setting, preferred_language: 'es')) }
+  let(:swedish_user) { FactoryBot.create(:user, user_setting: FactoryBot.create(:user_setting, preferred_language: 'sv')) }
+
   before do
     driven_by(:rack_test) # or :selenium if using a headless browser for JavaScript testing
   end
 
   it 'displays the form in English' do
-    visit new_contact_path(locale: 'en')
+    login_as(english_user, scope: :user)
```

Comment:
Is the `scope` necessary here?

Reply:
It's a devise thing: The `scope: :user ` tells Devise which user model to authenticate, and ensures the correct session and permissions are applied in tests with multiple user types. Because we only have one user, it is not necessary, but I thought it might be good practice to specify the scope explicitly for clarity and to future-proof the tests in case we have multiple user models later. I will take it out for now, though ☀️